scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -0,0 +1,64 @@
+package org.oppia.android.scripts.coverage
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import org.oppia.android.scripts.common.BazelClient
+import org.oppia.android.scripts.common.CommandExecutor
+import org.oppia.android.scripts.common.ScriptBackgroundCoroutineDispatcher
+import java.io.File
+
+/**
+ * Class responsible for running coverage analysis asynchronously.
+ *
+ * @param repoRoot the root directory of the repository
+ * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+ * @param commandExecutor Executes the specified command in the specified working directory
+ */
+class CoverageRunner(
+  private val repoRoot: File,
+  private val scriptBgDispatcher: ScriptBackgroundCoroutineDispatcher,
+  private val commandExecutor: CommandExecutor
+) {
+  private val bazelClient by lazy { BazelClient(repoRoot, commandExecutor) }
+
+  /**
+   * Runs coverage analysis asynchronously for the Bazel test target.
+   *
+   * @param bazelTestTarget Bazel test target to analyze coverage.
+   * @return a deferred value that contains the coverage data [will contain the proto for the coverage data].
+   */
+  fun runWithCoverageAsync(
+    bazelTestTarget: String
+  ): Deferred<String?> {
+    return CoroutineScope(scriptBgDispatcher).async {
+      val coverageDataFilePath = getCoverage(bazelTestTarget)
+      coverageDataFilePath
+    }
```

Comment:
```suggestion
    return CoroutineScope(scriptBgDispatcher).async { getCoverage(bazelTestTarget) }
```

Can be simplified a bit.

Also, perhaps something like 'retrieveCoverageResult' would be a better name for ``getCoverage`` (since it's not a simple getter).

Reply:
Got it! Done [[commit](https://github.com/oppia/oppia-android/pull/5423/commits/ee078e299a707ecdee4fbdc700ab2bc27360d900#diff-e1015bdd1aafa12e3e95de3b408525733c702992881ece1d2399b57ded6a44fdL46)]